eduid_am/tests/test_signup.py

```diff
@@ -29,39 +29,40 @@ def test_invalid_user(self):
             self.fetcher.fetch_attrs(bson.ObjectId('000000000000000000000000'))
 
     def test_existing_user_from_db(self):
-        self.maxDiff = None
+        fetched = self.fetcher.fetch_attrs(bson.ObjectId(M['_id']))
+
+        expected_passwords = [
+            {
+                'created_by': 'signup',
+                'credential_id': '112345678901234567890123',
+                'is_generated': False,
+                'salt': '$NDNv1H1$9c810d852430b62a9a7c6159d5d64c41c3831846f81b6799b54e1e8922f11545$32$32$',
+            }
+        ]
+        self.normalize_data(fetched['$set']['passwords'], expected_passwords)
+
+        expected_emails = [
+            {
+                'created_by': 'signup',
+                'email': 'johnsmith@example.com',
+                'primary': True,
+                'verified': True,
+                'verified_by': 'signup',
+            },
+            {'email': 'johnsmith2@example.com', 'primary': False, 'verified': True},
+            {'email': 'johnsmith3@example.com', 'primary': False, 'verified': False},
+        ]
+        self.normalize_data(fetched['$set']['mailAliases'], expected_emails)
+
         expected = {
             '$set': {
                 'eduPersonPrincipalName': 'hubba-bubba',
-                'mailAliases': [
-                    {
-                        'created_by': 'signup',
-                        'email': 'johnsmith@example.com',
-                        'primary': True,
-                        'verified': True,
-                        'verified_by': 'signup',
-                    },
-                    {'email': 'johnsmith2@example.com', 'primary': False, 'verified': True},
-                    {'email': 'johnsmith3@example.com', 'primary': False, 'verified': False},
-                ],
-                'passwords': [
-                    {
-                        'created_by': 'signup',
-                        'credential_id': '112345678901234567890123',
-                        'is_generated': False,
-                        'salt': '$NDNv1H1$9c810d852430b62a9a7c6159d5d64c41c3831846f81b6799b54e1e8922f11545$32$32$',
-                    }
-                ],
+                'mailAliases': expected_emails,
+                'passwords': expected_passwords,
             }
         }
-        res = self.fetcher.fetch_attrs(bson.ObjectId(M['_id']))
-        # remove the datetimes from the response,
-        # that carry their own tzinfo object from bson
-        del res['$set']['mailAliases'][0]['created_ts']
-        del res['$set']['mailAliases'][0]['verified_ts']
-        del res['$set']['passwords'][0]['created_ts']
 
-        assert res == expected
+        assert expected == fetched
```

Comment:
is it expected first and then actual, or the other way around? it matters when pytest formats the human readable error message, and you have the order the other way around in other asserts (like below on line 110)

Reply:
You are right